cmd/controller/ctrl.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"strconv"
+	"github.com/openebs/node-disk-manager/cmd/storage/block"
+	"github.com/openebs/node-disk-manager/cmd/types/v1"
+
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	apis "github.com/openebs/node-disk-manager/pkg/apis/ndm/v1alpha1"
+	clientset "github.com/openebs/node-disk-manager/pkg/client/clientset/versioned"
+)
+
+const (
+	NdmKind = "Disk"
+	NdmVersion = "openebs.io/v1alpha1"
+)
+
+// Controller is the controller implementation for do resources
+type Controller struct {
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	// sampleclientset is a clientset for our own API group
+	clientset clientset.Interface
+}
+
+// NewController returns a new sample controller
+func NewController(
+	kubeclientset kubernetes.Interface,
+	clientset clientset.Interface) *Controller {
+
+	controller := &Controller {
+		kubeclientset: kubeclientset,
+		clientset:     clientset,
+	}
+
+	return controller
+}
+
+// Run will discover the local disks and push them to the etcd
+func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
+	glog.Info("Started the controller")
+
+	err := PushDiskResources(c)
+	if err != true {
+		glog.Error("Unable to push disk objects to etcd ", err)
+	}
+	<-stopCh
+	glog.Info("Shutting down the controller")
+
+	return nil
+}
+
+func CreateDo(dr apis.Disk, c *Controller) {
```

Comment:
CreateDO

Reply:
Similarly other function nam3s. Also check if they can be private vs. public functions

Reply:
yeah, they can be private function. Chaged them.